packages/web/src/pages/sign-up-page/components/MobileNavHeader.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  Box,
+  Flex,
+  IconAudiusLogoHorizontal,
+  IconCaretLeft,
+  PlainButton,
+  iconSizes
+} from '@audius/harmony'
+import { useHistory } from 'react-router-dom'
+
+type MobileNavHeaderProps = {
+  isBackAllowed?: boolean
+}
+export const MobileNavHeader = (props: MobileNavHeaderProps) => {
+  const { isBackAllowed = true } = props
+  const history = useHistory()
+  const handleBackClick = () => {
+    history.goBack()
+  }
+  return (
+    <Flex
+      ph='xl'
+      pv='l'
+      w='100%'
+      css={(theme) => ({
+        backgroundColor: theme.color.background.white,
+        borderBottom: `solid 1px ${theme.color.border.default}`
+      })}
+      alignItems='center'
+      justifyContent='center'
+    >
+      <Flex
+        css={{ height: iconSizes.l, width: iconSizes.l }}
+        alignItems='center'
+      >
+        {isBackAllowed ? (
+          <PlainButton
+            css={{ padding: 0 }}
+            onClick={() => {
+              handleBackClick()
+            }}
+          >
+            <IconCaretLeft color='subdued' />
```

Comment:
wouldn't we want to do "iconLeft"? or youre saying this is the way when a plain-button just has icon? i think we should create an "icon-button" for these cases?

Reply:
I went ahead and batched all the icon feedback together into a ticket
https://linear.app/audius/issue/C-3385/harmony-icon-components-tweaks